tensorflow/python/ops/array_grad.py

```diff
@@ -354,9 +354,15 @@ def _PreventGradientGrad(op, _):
 def _GatherGrad(op, grad):
   """Gradient for Gather op."""
   # params can be large, so colocate the shape calculation with it.
+  #
+  # params can be super-large for sparse model, array_ops.shape raises
+  # exception inside Windows when any dimension is larger than int32,
```

Comment:
@fesun OK, sounds good. If you update the comment, this should be ready to merge.